src/vma/dev/cq_mgr_mlx5.cpp

```diff
@@ -775,7 +776,9 @@ void cq_mgr_mlx5::set_qp_rq(qp_mgr* qp)
 {
 	m_qp = static_cast<qp_mgr_eth_mlx5*> (qp);
 
-	m_qp->m_rq_wqe_counter = 0; /* In case of bonded qp, wqe_counter must be reset to zero */
+	if (typeid(qp_mgr_eth_direct) == typeid(*qp)) {
+		m_qp->m_rq_wqe_counter = 0; /* In case of bonded qp, wqe_counter must be reset to zero */
+	}
```

Comment:
I would prefer to use `dynamic_cast`.
```
qp_mgr_eth_mlx5* qp_ptr = dynamic_cast<qp_mgr_eth_mlx5*>(qp);
if (qp_ptr) {
	m_qp->m_rq_wqe_counter = 0; /* In case of bonded qp, wqe_counter must be reset to zero */
}
```

Note:
There is an important difference between the two methods:

if(A* test = dynamic_cast<A*>(&someClassCVar)) {
    // someClassCVar is A or publicly derived from A
}
Whereas:

if(typeid(someClassCVar) == typeid(A)) {
   // someClassCVar is of class A, not a derived class
}

It means that your check is specific for `qp_mgr_eth_direct` that is derived from `qp_mgr_eth_mlx5` nut it does not work for object having type `qp_mgr_eth_mlx5`.

Reply:
agree, It is ok for rivermax. changed.

Reply:
@oleg, @serglelv, we need to confirm with the reproducer that the issue is resolved still after this update.

Reply:
After speaking to @igor-ivanov I agree that we need to understand if it is at all valid that a QP which is not a `qp_mgr_eth_mlx5` or derived from it enters this code. And if not - we must understand why this happens and that's what me must fix.

@rafiw - please review!

Reply:
The stress test was passed using the fix.

Reply:
More comment:
1. `#include "qp_mgr_eth_direct.h"` is not needed
2. I think that fix should be done in following way
```
qp_mgr_eth_mlx5* qp_ptr = dynamic_cast<qp_mgr_eth_mlx5*>(qp);
if (qp_ptr) {
        m_qp = static_cast<qp_mgr_eth_mlx5*> (qp);
	m_qp->m_rq_wqe_counter = 0; /* In case of bonded qp, wqe_counter must be reset to zero */
}
```

But there is dangerous place as https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr_mp.cpp#L170

Reply:
what is the value of wqw_counter before you set it to 0?

Reply:
the value of counter is 0